presto-main/src/main/java/com/facebook/presto/operator/ShortArrayBlockEncodingBuffers.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import io.airlift.slice.SliceOutput;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.operator.ByteArrayUtils.writeLengthPrefixedString;
+import static com.google.common.base.Verify.verify;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.lang.Math.max;
+import static sun.misc.Unsafe.ARRAY_SHORT_INDEX_SCALE;
+
+public class ShortArrayBlockEncodingBuffers
```

Comment:
Why is `BlockEncodingBuffers` plural?  Should it be singular?

Reply:
Because each of them maintains a set of buffers, e.g. nullsBuffer, valuesBuffer, offsetsBuffer, etc. If most people think the plural form is confusing I'll rename it to sigular. cc @mbasmanova